### PR TITLE
Avocado Job Diff [V0]

### PR DESCRIPTION
**Pull request by apahim**
This is the initial version of Avocado Diff plugin. 
You will notice that some code is borrowed from other parts of Avocado (`if getattr(args, 'open_browser', False):`, `_setup_job()` and `_get_variants()`). Also, this is making intensive use of a/c/replay. Once we agree on the implementation, some renaming and refactoring are expected to take place so we can re-use the code instead of copy/paste.

### CODE REVIEW

File: docs/source/Diff.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     COMMAND LINE
+    -/usr/bin/avocado run sleeptest.py
+    +/usr/bin/avocado run passtest.py
+     
+     TOTAL TIME
+    -1.00251293182
+    +0.000292062759399
+     
+     VARIANTS
+     Variant 1: /
```
Comment (clebergnu) [inline docs/source/Diff.rst]: Wondering if non-differences should be included.

File: avocado/plugins/diff.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                setsid = getattr(os, 'setsid', None)
+                if not setsid:
+                    setsid = getattr(os, 'setpgrp', None)
+                inout = file(os.devnull, "r+")
+                cmd = ['xdg-open', tmp_file.name]
+                subprocess.Popen(cmd, close_fds=True, stdin=inout, stdout=inout,
+                                 stderr=inout, preexec_fn=setsid)
+
+    def _valid_filters(self, string):
+        return string.split(',')
```
Comment (apahim) [inline avocado/plugins/diff.py]: Ops. This will be populated in v1 :)

File: docs/source/Diff.rst (hunk 1 above)
Comment (apahim) [inline docs/source/Diff.rst]: It's a way to go. I wonder how we should identify the section if non-differences disappear. Maybe: 
```diff
~$ avocado diff 7025aaba9c2 384b949c
--- 7025aaba9c2ab8b4bba2e33b64db3824810bb5df
+++ 384b949c991b8ab324ce67c9d9ba761fd07672ff
@@ -3 +3 @@
-COMMAND LINE: /usr/bin/avocado run sleeptest.py
+COMMAND LINE: /usr/bin/avocado run passtest.py
@@ -6 +6 @@
-TOTAL TIME: 1.00251293182
+TOTAL TIME: 0.000292062759399
@@ -12 +12 @@
-TEST RESULTS: 1-sleeptest.py:SleepTest.test: PASS
+TEST RESULTS: 1-passtest.py:PassTest.test: PASS
```
Or even:

```diff
$ avocado diff 7025aaba9c2 384b949c
--- 7025aaba9c2ab8b4bba2e33b64db3824810bb5df
+++ 384b949c991b8ab324ce67c9d9ba761fd07672ff
@@ -3 +3 @@
-COMMAND_LINE_1
+COMMAND_LINE_2
-/usr/bin/avocado run sleeptest.py
+/usr/bin/avocado run passtest.py
@@ -6 +6 @@
-TOTAL_TIME_1
+TOTAL_TIME_2
-1.00251293182
+0.000292062759399
@@ -12 +12 @@
-TEST_RESULTS_1
+TEST_RESULTS_2
-1-sleeptest.py:SleepTest.test: PASS
+1-passtest.py:PassTest.test: PASS
```

Comment (apahim) [inline docs/source/Diff.rst]: Yet another option:
```diff
$ avocado diff 7025aaba9c2 384b949c
--- a/command_line
+++ b/command_line
@@ -1 +1 @@
-/usr/bin/avocado run sleeptest.py
+/usr/bin/avocado run passtest.py

--- a/total_time
+++ b/total_time
@@ -1 +1 @@
-1.00251293182
+0.000292062759399

--- a/test_results
+++ b/test_results
@@ -1 +1 @@
-1-sleeptest.py:SleepTest.test: PASS
+1-passtest.py:PassTest.test: PASS
```
@adereis, any preference?

Comment (adereis) [inline docs/source/Diff.rst]: Simply filter-out the sections which are the same, just as if one was manually using --diff-filter. The result will be like this:

--- aa416ece963381fb427431002aabea7aa99e3f4f
+++ 166a527461944518899352f1cfd88b36bb6d45f1
@@ -1,13 +1,12 @@
 
 COMMAND LINE
-./scripts/avocado run /bin/true /bin/false /bin/true /bin/false passtest.py
+./scripts/avocado run passtest.py failtest.py /bin/true /bin/false
 
 TOTAL TIME
-0.03422498703
+0.0232672691345
 
 TEST RESULTS
-1-/bin/true: PASS
-2-/bin/false: FAIL
+1-passtest.py:PassTest.test: PASS
+2-failtest.py:FailTest.test: FAIL
 3-/bin/true: PASS
 4-/bin/false: FAIL
-5-passtest.py:PassTest.test: PASS

Comment (adereis) [inline docs/source/Diff.rst]: Context is important in a unified diff. What is distracting are the sections where there are no differences.

Comment (apahim) [inline docs/source/Diff.rst]: ok

**Pull request closed without merging**